web/src/utils/icmp.ts

```diff
@@ -1,75 +1,100 @@
+  { value: 12, name: 'ICMP_HOST_UNR_TOS' },
+  { value: 13, name: 'ICMP_PKT_FILTERED', description: 'Packet filtered' },
+  { value: 14, name: 'ICMP_PREC_VIOLATION', description: 'Precedence violation' },
+  { value: 15, name: 'ICMP_PREC_CUTOFF', description: 'Precedence cut off' }
+  //{ value: 15, name: 'NR_ICMP_UNREACH', description: 'instead of hardcoding immediate value' },
+] as const;
 
-export const getType = (v?: number): string => {
-  if (!v) {
-    return '';
-  }
-  return ICMP_TYPES[v];
-};
+const icmpUnreachCodesValues = ICMP_UNREACH_CODES.map(v => v.value);
+export type ICMP_UNREACH_CODES_VALUES = typeof icmpUnreachCodesValues[number];
+
+const icmpUnreachCodesNames = ICMP_UNREACH_CODES.map(v => v.name);
+export type ICMP_UNREACH_CODES_NAMES = typeof icmpUnreachCodesNames[number];
+
+export const ICMP_REDIRECT_CODES: ReadOnlyValues = [
+  { value: 0, name: 'ICMP_REDIR_NET', description: 'Redirect Net' },
+  { value: 1, name: 'ICMP_REDIR_HOST', description: 'Redirect Host' },
+  { value: 2, name: 'ICMP_REDIR_NETTOS', description: 'Redirect Net for TOS' },
+  { value: 3, name: 'ICMP_REDIR_HOSTTOS', description: 'Redirect Host for TOS' }
+] as const;
+
+const icmpRedirectCodesValues = ICMP_REDIRECT_CODES.map(v => v.value);
+export type ICMP_REDIRECT_CODES_VALUES = typeof icmpRedirectCodesValues[number];
+
+const icmpRedirectCodesNames = ICMP_REDIRECT_CODES.map(v => v.name);
+export type ICMP_REDIRECT_CODES_NAMES = typeof icmpRedirectCodesNames[number];
+
+export const ICMP_TIME_EXCEEDED_CODES: ReadOnlyValues = [
+  { value: 0, name: 'ICMP_EXC_TTL', description: 'TTL count exceeded' },
+  { value: 1, name: 'ICMP_EXC_FRAGTIME', description: 'Fragment Reass time exceeded' }
+] as const;
+
+const icmpTimeExceededCodesValues = ICMP_TIME_EXCEEDED_CODES.map(v => v.value);
+export type ICMP_TIME_EXCEEDED_CODES_VALUES = typeof icmpTimeExceededCodesValues[number];
+
+const icmpTimeExceededCodesNames = ICMP_TIME_EXCEEDED_CODES.map(v => v.name);
+export type ICMP_TIME_EXCEEDED_CODES_NAMES = typeof icmpTimeExceededCodesNames[number];
+
+export type ICMP_CODES_VALUES =
+  | ICMP_UNREACH_CODES_VALUES
+  | ICMP_REDIRECT_CODES_VALUES
+  | ICMP_TIME_EXCEEDED_CODES_VALUES;
 
-export const getCode = (t?: number, v?: number): string => {
-  if (!t || !v) {
-    return '';
+export const getICMPCode = (t?: ICMP_TYPES_VALUES, c?: ICMP_CODES_VALUES): ReadOnlyValue | undefined => {
+  if (!t || !c) {
+    return undefined;
   }
   switch (t) {
-    case ICMP_TYPES.ICMP_DEST_UNREACH:
-      return ICMP_UNREACH_CODES[v];
-    case ICMP_TYPES.ICMP_REDIRECT:
-      return ICMP_REDIRECT_CODES[v];
-    case ICMP_TYPES.ICMP_TIME_EXCEEDED:
-      return ICMP_TIME_EXCEEDED_CODES[v];
+    case 3: // ICMP_DEST_UNREACH:
+      return ICMP_UNREACH_CODES.find(v => v.value === c);
+    case 5: // ICMP_REDIRECT:
+      return ICMP_REDIRECT_CODES.find(v => v.value === c);
+    case 11: // ICMP_TIME_EXCEEDED:
+      return ICMP_TIME_EXCEEDED_CODES.find(v => v.value === c);
```

Comment:
what about icmp6 codes and types they aren't the same as in icmp4 ?

Reply:
Correct:
https://www.iana.org/assignments/icmp-parameters/icmp-parameters.xhtml#icmp-parameters-types
https://www.iana.org/assignments/icmpv6-parameters/icmpv6-parameters.xhtml#icmpv6-parameters-2

I should implement both and switch according to for protocol field

```
1 | ICMP | Internet Control Message
vs 
58 | IPv6-ICMP | ICMP for IPv6
```

Reply:
https://github.com/netobserv/network-observability-console-plugin/pull/362/commits/eb1c5be0f8207647eaed45f581a6862131f6838f